### ISSUE TITLE
dotnet add package doesn't update existing packages when Central Package Management is enabled

### ISSUE THREAD

**Issue by BenjaminBatman** [NuGet/Home#12769]
### NuGet Product Used

dotnet.exe

### Product Version

.Net SDK 7.0.3xxx 

### Impact

It's more difficult to complete my work

### Repro Steps & Context

Centeral Package Management is enabled on my solution and I am trying to update an existing package in the Directory.Packages.props file to the newest version using `dotnet add package`. However when running `dotnet add [PROJECT] package [PACKAGE]`, the package never updates.

1. Add a Nuget package to the Directory.Packages.props file that is out dated.
2. Run `dotnet add [PROJECT] package [PACKAGE]`
3. Checking the Directory.Package.props file, the package will be the same version it was before running `dotnet add package`.

I have tried a few different tests. It seems to work if you attach the version to it. `dotnet add [PROJECT] package [PACKAGE] -v [VERSION]`. It also seems to correctly add the package if it didn't exist in the Directory.Packages.props file previously.

**Comment by martinrrm** [comment]
@BenjaminBatman Hi, thanks for filling this issue. looking at the design for `dotnet add` in CPM projects https://github.com/NuGet/Home/blob/dev/proposed/2022/cpm-dotnet-cli-add-package-support.md#explanation and in scenario 11, dotnet add [PROJECT] package [PACKAGE] we considered that the best option was No-op. We decided it would be best to only update a version when the user explicitly passes in the command line argument.

**Comment by BenjaminBatman** [comment]
@martinrrm This makes sense. I could see if it is being used in a lot of projects in the solution, you might not want to accidently update it to the newest for one project and mess something up in the another. Would it be possible to update it to accept the version parameter with no value `-v` or a value of latest `-v latest` to force it to update to the latest package version?

**Comment by IanKemp** [comment]
Working link: https://github.com/NuGet/Home/blob/dev/accepted/2022/cpm-dotnet-cli-add-package-support.md#explanation

> We decided it would be best to only update a version when the user explicitly passes in the command line argument.

That would be fine if the output of `dotnet add package` explained this, but it does not; instead it just does nothing and silently exists, leaving the user sitting there wondering WTF is going on... did they make a mistake? Is there a bug in a `dotnet package add`? Nah, that's not possible, they must have made a mistake... cue much time wasted finding out they did not, in fact, make a mistake, because the only place this behaviour is documented is in a random file in the NuGet repo, not in a more useful and accessible location like, I dunno, [the official docs page](https://learn.microsoft.com/dotnet/core/tools/dotnet-add-package). I can personally attest that this is an abominably bad user experience.

But over and above that, this decision makes absolutely no sense; why should the behaviour differ between CPM and non-CPM scenarios? Because of unintended consequences? **It's not your place to make that decision**, it's for the person running the command to understand what it's going to do before they choose to run it, and that's regardless of whether they explicitly specify a version or not. [I did not stutter](https://knowyourmeme.com/memes/did-i-stutter-stanley-the-office) when I typed and executed that command.

In short, the current behaviour is a bug created by the NuGet team's own decision to do something completely nonsensical, and as such should be fixed - especially in a world where CPM is (finally) taking over.

**Comment by cz-dev-ge** [comment]
Maybe we could simply get another command "update" or "upgrade".

**Comment by grotkiv** [comment]
> Maybe we could simply get another command "update" or "upgrade".

I would appreciate an update command that updates all package versions in the `Directory.Package.props` file. In the event of any issues, it could be easily reverted using Git.

**Comment by zivkan** [comment]
> Maybe we could simply get another command "update" or "upgrade".

* https://github.com/NuGet/Home/issues/4103
